### PR TITLE
PER-392: create_byte_pairs_for_caption_string now keeps track of the …

### PR DESCRIPTION
**Pull request by kagutaba256**
…number of basic chars being even or odd, and appends 0x80 if a special char is put in or if the caption string comes to an end

### CODE REVIEW

File: src/server/cea_608_encoder/caption_string_utility.py [hunk 1]
```diff
@@ -107,17 +107,33 @@ def create_byte_pairs_for_caption_string(caption_string: str, channel_toggle: in
...
+            # and if so, append 0x80
+            if basic_chars_is_odd:
+                byte_list.append(hex(get_single_null_byte_with_parity()))
             if check_parity(first_byte) == 0:
                 first_byte = add_parity_to_byte(first_byte)
             byte_list.append(hex(first_byte))
+            # since we know we just appended a full bytepair, we set to false.
+            basic_chars_is_odd = False
+        else:
+            basic_chars_is_odd = ~basic_chars_is_odd
```
Comment (Markobozic) [inline src/server/cea_608_encoder/caption_string_utility.py]: i am 95% sure this is not doing what i think it should be doing

Comment (Markobozic) [reply src/server/cea_608_encoder/caption_string_utility.py]: if you're trying to negate basic_chars that is

Comment (kagutaba256) [reply src/server/cea_608_encoder/caption_string_utility.py]: i tested it in a shell, it actually does work just fine- when i print after negating, it turns false into 0 and true into 1. i can totally change it

Comment (kagutaba256) [reply src/server/cea_608_encoder/caption_string_utility.py]: i thought this would be best for readability, didnt know it would be weird.

Comment (Markobozic) [reply src/server/cea_608_encoder/caption_string_utility.py]: Screen Shot 2019-10-31 at 6 03 19 PM

**Comment by kagutaba256** [comment]
now fixed with latest commit

**Review by Markobozic — APPROVED** [review]

**Pull request merged**